Add alert manager

MarcelMue opened the pull request:
This PR is to add Alertmanager templates and Rules:
- [x] Add Alertmanager templates
- [x] Add some sample Prometheus rules
- [x] Make sure Prometheus is running and shows rules and alert status (in WebUI)
- [x] Make sure Prometheus Pod is actually reaching Alertmanager Pod (can't see anything in logs)
- [x] Make sure Alerts actually arrive in a Slack channel 
- [x] Update Readme/Documentation

(Implements parts of [#1263](https://github.com/giantswarm/giantswarm/issues/1263) )

MarcelMue commented:
Tested in a slack channel, alerting and repeated alerting worked with the default config.

I am however unable to access any Alertmanager UI, will investigate why.

webwurst commented on build.sh:
```diff
@@ -0,0 +1,52 @@
+for file in $(find ./manifests -type f -name "*.yaml" | sort) ; do
+  echo "---" >> "$target"
+  cat "$file" >> "$target"
+done
```
In general I would like to get rid of the `/configs` directory. It seemed to be a good idea once, but might be more confusing then helping in the end. Just for your info, nothing wrong with your code.

MarcelMue replied:
I agree, I opened #67 to fix this. I think a new PR should be made to work on this.

webwurst commented on manifests/alertmanager/alertmanager-templates.yaml:
```diff
@@ -0,0 +1,205 @@
+    {{ end }}
+kind: ConfigMap
+metadata:
+  creationTimestamp: null
```
Can be removed.

MarcelMue replied:
These timestamps appear because this file was built from files in the `/configs`directory. So they will reappear every time we create them.

I think removing `/configs` should solve this issue.

webwurst commented on manifests/alertmanager/configmap.yaml:
```diff
@@ -0,0 +1,83 @@
+      - match:
+          severity: email
+        receiver: slack_alert
+    #   receiver: email_alert
```
Should this be `receiver: email_alert` in the end?

MarcelMue replied:
Yes, I replaced it with `email_alerts`

webwurst commented on manifests/alertmanager/deployment.yaml:
```diff
@@ -0,0 +1,48 @@
+        args:
+          - '-config.file=/etc/alertmanager/config.yml'
+          - '-storage.path=/alertmanager'
+          - '-web.external-url=$(EXTERNAL_URL)/alertmanager'
```
This might not be needed in the most cases, if behind an `ingress` at root level. So I would leave that out for now.

webwurst commented on manifests/alertmanager/deployment.yaml:
```diff
@@ -0,0 +1,48 @@
+    spec:
+      containers:
+      - name: alertmanager
+        image: prom/alertmanager:latest
```
We could update the image to `quay.io/prometheus/alertmanager:v0.7.0`. Seems to be the official registry for the Prometheus project now.

puja108 replied:
+1, and not use latest

webwurst replied:
And now there is `v0.7.1` already ;) https://github.com/prometheus/alertmanager/releases/tag/v0.7.1

MarcelMue replied:
I bumped the version to `0.7.1` and changed the repo.

webwurst commented on manifests/alertmanager/service.yaml:
```diff
@@ -0,0 +1,19 @@
+spec:
+  selector:
+    app: alertmanager
+  #type: NodePort
```
`NodePort` is needed for Minikube as long as no ingress is setup. I could access the Web-UI via the NodePort after disabling `-web.external-url`.

puja108 replied:
yes, and it doesn't hurt

MarcelMue replied:
I removed `EXTERNAL_URL` and `NodePort` is now in use again.

RahulMahale commented:
I am yet to test this PR, But looks like an alertmanager templates are inspiration from https://github.com/kayrus/prometheus-kubernetes

puja108 commented:
@RahulMahale you are right, his/hers repo is actually inspired by ours, and our altermanager is inspired by his/hers.

@MarcelMue we should note that somewhere in the Readme or so.

MarcelMue commented:
@puja108 @RahulMahale I added a `Credit`section to the `README` and referred to Kayrus' repository there.

Also added some very brief explanation to `docs` and bumped Prometheus version to `1.7.0`

RahulMahale commented:
@MarcelMue  thats great, I have used those alertmanager remplates from Kayrus' repository  and that worked great with this repository. 

Thank you all for organizing this deployment templates :1st_place_medal:
